### PR TITLE
[REVIEW][FEATURE][GAWE-63]-Create Screen Instruction Reset

### PR DESCRIPTION
**Pull request by azkifairuz**
#### Description

>Screen Instruction Reset.

#### Ticket Reference(s)

[GAWE-64](https://bluehabit.atlassian.net/browse/GAWE-63)

##### UI Changes

>

##### Coverage

>  gambar

N/A

#### Note(s)

> Note
masih bingung gima buat nge direct ke email

[GAWE-64]: https://bluehabit.atlassian.net/browse/GAWE-64?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by triandamai** [comment]
Judul PR nya benerin dulu ya

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import androidx.compose.ui.unit.sp
+import app.trian.mvi.ui.extensions.from
+import com.bluehabit.core.ui.theme.GaweTheme
+import com.bluehabit.core.ui.theme.Gray900
+import com.bluehabit.core.ui.theme.Primary600
+
+@Composable
+fun InstructionResetScreen(
+    openEmail: () -> Unit = {},
+    tryAgain: () -> Unit = {},
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:42]: ganti jadi `onRetry` aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import app.trian.mvi.ui.extensions.from
+import com.bluehabit.core.ui.theme.GaweTheme
+import com.bluehabit.core.ui.theme.Gray900
+import com.bluehabit.core.ui.theme.Primary600
+
+@Composable
+fun InstructionResetScreen(
+    openEmail: () -> Unit = {},
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:41]: ganti jadi `onOpenEmailApp` aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    Column(
+        modifier = Modifier
+            .fillMaxHeight()
+            .padding(
+                vertical = 28.dp.from(context = context),
+                horizontal = 26.dp.from(context = context),
+            )
+            .background(Color.White),
+
+        verticalArrangement = Arrangement.spacedBy(95.dp.from(context = context)), horizontalAlignment = Alignment.CenterHorizontally
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:54]: horizontal alignment nya enter aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                Text(
+                    text = "Instruksi Terkirim!",
+                    style = MaterialTheme.typography.h6,
+                    lineHeight = 30.sp,
+                    fontWeight = FontWeight.W600,
+                    textAlign = TextAlign.Center,
+                    color = Gray900
+                )
+                Text(
+                    text = "Cek email anda dan klik link pada instruksi tersebut.",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:84]: extract ke string resource aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                    id = com.bluehabit.core.ui.R.drawable.gawean_logo
+                ),
+                contentDescription = "logo"
+            )
+            Column(
+                verticalArrangement = Arrangement.spacedBy(8.dp.from(context = context)),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Text(
+                    text = "Instruksi Terkirim!",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:76]: extract ke string resource aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                onClick = openEmail, modifier = Modifier
+                    .width(320.dp)
+                    .height(48.dp)
+                    .background(
+                        color = Color(0xFF7F56D9),
+                        shape = RoundedCornerShape(size = 8.dp)
+                    )
+            ) {
+                Text(
+                    text = "Buka Email",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:117]: extract ke string resource aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                    color = Color(0xFFFFFFFF)
+                )
+            }
+            Column(
+                verticalArrangement = Arrangement.spacedBy(8.dp.from(context = context)),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Text(
+                    text = "Tidak menerima email? Coba cek spam",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:126]: extract ke string resource aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                    style = MaterialTheme.typography.caption,
+                    lineHeight = 20.sp,
+                    fontWeight = FontWeight.W400,
+                    textAlign = TextAlign.Center,
+                    color = Gray900
+                )
+                Text(
+                    text = "atau",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:134]: extract ke string resource aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+                    lineHeight = 20.sp,
+                    fontWeight = FontWeight.W400,
+                    textAlign = TextAlign.Center,
+                    color = Gray900
+                )
+                Text(
+                    modifier = Modifier.clickable {
+                        tryAgain()
+                    },
+                    text = "coba lagi dengan alamat email yang lain.",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:145]: extract ke string resource aja

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+                    color = Primary600
+                )
+            }
+        }
+    }
+}
+
+@Composable
+@Preview(widthDp = 384, heightDp = 854)
+fun DefaultPreview() {
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:159]: Hindari pake Default Preview namingnya pasti bentrok sama yang lain nanti
misal 

```kotlin
fun PreviewInstructionScreen()
```

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+                )
+            }
+            Column(
+                Modifier.padding(30.dp)
+            ) {
+                Image(
+                    painter = painterResource(
+                        id = com.bluehabit.core.ui.R.drawable.hero_intruction_reset_password
+                    ),
+                        contentDescription = "logo"
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:99]: formatiing ulang masih blm rapih

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+                ),
+            verticalArrangement = Arrangement.spacedBy(24.dp.from(context = context)),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+
+            Image(
+                painter = painterResource(
+                    id = com.bluehabit.core.ui.R.drawable.gawean_logo
+                ),
+                contentDescription = "logo"
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:69]: contetn descriptionnya di perjelas lagi supaya ada maknanya

**Review by triandamai — CHANGES_REQUESTED** [review]
revisi

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt (hunk 2 above)
Comment (azkifairuz) [reply feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:41]: done

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt (hunk 3 above)
Comment (azkifairuz) [reply feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt]: done

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt (hunk 4 above)
Comment (azkifairuz) [reply feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt]: done

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt (hunk 5 above)
Comment (azkifairuz) [reply feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt]: done

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+                onClick = onOpenEmailApp, modifier = Modifier
+                    .width(320.dp)
+                    .height(48.dp)
+                    .background(
+                        color = Color(0xFF7F56D9),
+                        shape = RoundedCornerShape(size = 8.dp)
+                    )
+            ) {
+                Text(
+                    text = "Buka Email",
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/resetPassword/screen/InstructionResetScreen.kt:116]: Masih blm di extract nih

**Review by triandamai — CHANGES_REQUESTED** [review]
Cek cek

**Review by triandamai — APPROVED** [review]

**Pull request merged**